conn.go

```diff
@@ -378,20 +379,35 @@ func (c *conn) Err() error {
 	return &ConnectionError{inner: c.err}
 }
 
+func (c *conn) NewSession() (*Session, error) {
+	c.sessionsByChannelMu.Lock()
+	defer c.sessionsByChannelMu.Unlock()
+
+	// create the next session to allocate
+	// note that channel always start at 0
+	channel, ok := c.channels.Next()
+	if !ok {
+		return nil, fmt.Errorf("reached connection channel max (%d)", c.channelMax)
+	}
+	session := newSession(c, uint16(channel))
+	ch := session.channel
+	c.sessionsByChannel[ch] = session
```

Comment:
NIT: We could just use 'channel' and avoid the intermediary 'ch'.

Reply:
Fixed